scipy/stats/stats.py

```diff
@@ -3692,6 +3696,230 @@ def f_oneway(*args, axis=0):
+    Brand Comparison    Mean Difference     Simultanious 95% Confidence Limits
+    2 - 3               4.340               0.691           7.989   ***
+    2 - 1               4.600               0.951           8.249   ***
+    3 - 2              -4.340              -7.989          -0.691   ***
+    3 - 1               0.260              -3.389           3.909
+    1 - 2              -4.600              -8.249          -0.951   ***
+    1 - 3              -0.260              -3.909           3.389
+
+    Let's test our function against the SAS result:
+    group1 = [24.5, 23.5, 26.4, 27.1, 29.9]
+    group2 = [28.4, 34.2, 29.5, 32.2, 30.1]
+    group3 = [26.1, 28.3, 24.3, 26.2, 27.8]
+    args=[group1,group2,group3]
+    tukeykramer(*args)
+    ~~~~~~~~~~~~
+    Comparison between groups with Simultanious
+    with 95% Confidence Limits and significance at the 0.05 level)
+    ~~~~~~~~~~~~
+    group     mean      min      max      sig
+    1 - 2    -4.600    -8.249  -0.951     1.0
+    1 - 3    -0.260    -3.909   3.389     0.0
+    2 - 1     4.600     0.951   8.249     1.0
+    2 - 3     4.340     0.691   7.989     1.0
+    3 - 1     0.260    -3.389   3.909     0.0
+    3 - 2    -4.340    -7.989  -0.691     1.0
+
+    # second just different sizes
+    group1 = [24.5, 23.5, 26.28, 26.4, 27.1, 29.9, 30.1, 30.1]
+    group2 = [28.4, 34.2, 29.5, 32.2, 30.1]
+    group3 = [26.1, 28.3, 24.3, 26.2, 27.8]
+    tukeykramer(group1, group2, group3)
+
+    expected from SAS:
+    Brand Comparison    Mean Difference     Simultanious 95% Confidence Limits
+    2 - 3               3.645               0.268        	7.022   ***
+    2 - 1               4.34                0.593        	8.087   ***
+    3 - 2              -3.645              -7.022      	   -0.268   ***
+    3 - 1               0.695              -2.682         	4.072
+    1 - 2              -4.34               -8.087	       -0.593   ***
+    1 - 3              -0.695              -4.072	        2.682
+
+    actual:
+    ~~~~~~~~~~~~
+    Comparison between groups with Simultanious
+    (with 95% Confidence Limits and significance at the 0.05 level)
+    ~~~~~~~~~~~~
+    group     mean      min      max      sig
+    1 - 2    -3.645    -7.022  -0.268     1.0
+    1 - 3     0.695    -2.682   4.072     0.0
+    2 - 1     3.645     0.268   7.022     1.0
+    2 - 3     4.340     0.593   8.087     1.0
+    3 - 1    -0.695    -4.072   2.682     0.0
+    3 - 2    -4.340    -8.087  -0.593     1.0
+
+    """
+    args = [np.asarray(arg) for arg in args]
+    means = [np.mean(arg) for arg in args]
+
+    def get_q(p, r, v, guess=3):
+        def wrapper_cdf(q_crit, treatments, ddof, alpha):
+            return alpha - (1 - prtrng(q_crit, ddof,
+                                       treatments)[0])
+        res = root(wrapper_cdf, guess, args=(r, v, p))
+        if res.success:
+            return res.x.item()
+        else:
+            raise ValueError(res.message)
```

Comment:
@DominicChm

```suggestion
    def get_q(p, r, v, guess=3):
        def wrapper_cdf(q_crit, treatments, ddof, alpha):
            return alpha - (1 - prtrng(q_crit, ddof,
                                       treatments)[0])
        res = root_scalar(wrapper_cdf, guess, args=(r, v, p))
        if res.success:
            return res.x.item()
        else:
            raise ValueError(res.message)
```

Reply:
Also use math behind `QTRNG0` for a guess, and we should try to determine bounds.